homeassistant/components/homekit/accessories.py

```diff
@@ -0,0 +1,55 @@
+"""Extend the basic Accessory and Bridge functions."""
+from pyhap.accessory import Accessory, Bridge, Category
+
+from .const import (
+    SERVICES_ACCESSORY_INFO, MANUFACTURER,
+    CHAR_MODEL, CHAR_MANUFACTURER, CHAR_SERIAL_NUMBER)
+
+
+class HomeAccessory(Accessory):
+    """Class to extend the Accessory class."""
+
+    ALL_CATEGORIES = Category
+
+    def __init__(self, display_name):
+        """Initialize a Accessory object."""
+        super().__init__(display_name)
+
+    def set_category(self, category):
+        """Set the category of the accessory."""
+        self.category = category
+
+    def set_services(self, services):
+        """Define the services to be available for the accessory."""
+        from pyhap.loader import get_serv_loader
+        self.add_service(get_serv_loader().get(services))
+
+    def set_accessory_info(self, model, manufacturer=MANUFACTURER,
+                           serial_number='0000'):
+        """Set the default accessory information."""
+        service_info = self.get_service(SERVICES_ACCESSORY_INFO)
+        service_info.get_characteristic(CHAR_MODEL) \
+                    .set_value(model, False)
```

Comment:
You can safely do `set_value(value)` without the False from version 1.1.3 or 1.1.4.

Reply:
Does this method do I/O ?

Reply:
Short answer: no.
Long: Depends, if the containing accessory is running - yes, otherwise no (at the point above, the Acc is getting initialised, so no). If you **don't** want IO when setting a value, you can do `set_value(value, should_notify=False)`.